src/components/Layout/Layout.tsx

```diff
@@ -1,82 +1,90 @@
-import { useLocalStorage } from 'react-use'
 
 import { useListenDeepLinks } from '@/hooks/useListenDeepLinks'
 
+import {
+  ImperativePanelHandle,
+  Panel,
+  PanelGroup,
+  PanelResizeHandle,
+} from '../ResizablePanel'
+
 import { ActivityBar } from './ActivityBar'
 import { Sidebar } from './Sidebar'
 
 export function Layout() {
-  const [isSidebarExpanded, setIsSidebarExpanded] = useLocalStorage(
-    'isSidebarExpanded',
-    true
-  )
   const location = useLocation()
-  useListenDeepLinks()
 
-  const handleVisibleChange = (index: number, visible: boolean) => {
-    if (index !== 1) return
-    setIsSidebarExpanded(visible)
-  }
+  const [sidebar, setSidebar] = useState<ImperativePanelHandle | null>(null)
+  const [isSidebarExpanded, setIsSidebarExpanded] = useState(true)
+
+  useListenDeepLinks()
 
   useEffect(() => {
     window.studio.app.changeRoute(location.pathname)
   }, [location])
 
   return (
-    <Box
+    <Flex
       height="100dvh"
       position="relative"
       css={css`
-        /* Allotment */
+        /* Allotment - need to keep this until fully migrated */
         --focus-border: var(--accent-9);
         --separator-border: var(--gray-5);
         --sash-hover-size: 2px;
       `}
     >
-      {!isSidebarExpanded && (
-        <IconButton
-          size="1"
-          variant="ghost"
-          color="gray"
-          onClick={() => setIsSidebarExpanded(true)}
-          css={css`
-            position: absolute;
-            top: var(--space-4);
-            left: 64px;
-            transform: translateX(-50%);
-            z-index: 10000;
-          `}
-        >
-          <PanelLeftOpenIcon />
-        </IconButton>
-      )}
-      <Allotment onVisibleChange={handleVisibleChange}>
-        <Allotment.Pane minSize={64} maxSize={64}>
-          <ActivityBar />
-        </Allotment.Pane>
-        <Allotment.Pane
-          minSize={200}
-          maxSize={300}
-          preferredSize={280}
-          visible={isSidebarExpanded}
-          snap
+      <div
+        css={css`
+          position: relative;
+          min-width: 64px;
+          border-right: 1px solid var(--gray-5);
+        `}
+      >
+        <ActivityBar />
+        {!isSidebarExpanded && (
+          <IconButton
+            size="1"
+            variant="ghost"
+            color="gray"
+            css={css`
+              position: absolute;
+              top: var(--space-4);
+              right: calc(var(--space-2) * -1);
+              background-color: var(--color-background);
+              z-index: var(--layer-expand-button);
+            `}
+            onClick={() => sidebar?.expand()}
+          >
+            <PanelLeftOpenIcon />
+          </IconButton>
+        )}
+      </div>
+      <PanelGroup direction="horizontal" autoSaveId="main-layout">
+        <Panel
+          ref={setSidebar}
+          collapsible
+          defaultSize={15}
+          minSize={11}
+          maxSize={16}
```

Comment:
Apparently these are in % and there's [no official way ](https://github.com/bvaughn/react-resizable-panels/tree/main/packages/react-resizable-panels#can-panel-sizes-be-specified-in-pixels) to set limits in px.
I think it's a noticeable issue, but is only a big deal for the sidebar:
<img width="264" height="191" alt="Image" src="https://github.com/user-attachments/assets/2341ecae-be5e-443d-8b4c-201ed13e6b65" />